pkg/client/record/event.go

```diff
@@ -145,7 +145,11 @@ func recordEvent(sink EventSink, event *api.Event, updateExistingEvent bool) boo
 	var err error
 	if updateExistingEvent {
 		newEvent, err = sink.Update(event)
-	} else {
+	}
+	// Update can fail because the event may have been removed and it no longer exists.
+	if !updateExistingEvent || (updateExistingEvent && err != nil) {
+		// Making sure that ResourceVersion is empty on creation
```

Comment:
Does it make sense to fall back to create on all update failures? Or should we only fall back to create on "Key not found" failures?